source/acquaman/AMScanActionControllerBasicFileWriter.cpp

```diff
@@ -78,8 +78,17 @@ void AMScanActionControllerBasicFileWriter::writeToFile(int fileRank, const QStr
 		qDebug() << "Stream (1) writing differential " << startTime.msecsTo(endTime) << " for size " << textToWrite.size();
 		*/
 
+		QFileInfo dataFileInfo(filePath_+".dat");
+		int sizeBeforeWrite = dataFileInfo.size();
+
 		AMTextStream dataStream(dataFile_);
 		dataStream << textToWrite;
+		dataStream.flush();
```

Comment:
I think @XuetaoLiu-CLS is "right" here. The device in question is a QFile (dataFile_), so I think that the AMTextStream (QTextStream) is "operating" on that. Being said, it's plain that under normal circumstances we are writing things to the file, so we don't "need" to call flush() to write anything. Rather, it's most likely that calling flush() forces the data to be flushed so that when @XuetaoLiu-CLS is checking before and after he gets a different number. I imagine that under normal circumstances, the OS decides when to actually flush the device optimally (I hope that we're not going to significantly decrease performance by doing it this way). @dretrex @XuetaoLiu-CLS @iainworkman can you guys think of a way of refactoring this that doesn't rely on flushing the buffer in line?